omnibus/config/software/datadog-agent-integrations.rb

```diff
@@ -145,54 +131,56 @@
       end
     end
 
+    # Render the filtered requirements file
     erb source: "static_requirements.txt.erb",
-        dest: "#{static_reqs_filtered_file}",
+        dest: "#{static_reqs_out_file}",
         mode: 0640,
         vars: { requirements: requirements }
 
-    # Install the static environment requirements that the Agent and all checks will use
+    # Use pip-compile to create the final requirements file. Notice when we invoke `pip` through `python2 -m pip <...>`,
+    # there's no need to refer to `pip2`, the interpreter will pick the right script.
     if windows?
-      command("#{python_bin} -m #{python_pip_no_deps}\\datadog_checks_base")
-      command("#{python_bin} -m #{python_pip_no_deps}\\datadog_checks_downloader --install-option=\"--install-scripts=#{windows_safe_path(install_dir)}/bin\"")
-      command("#{python_bin} -m piptools compile --generate-hashes --output-file #{windows_safe_path(install_dir)}\\#{agent_requirements_file} #{static_reqs_filtered_file}")
+      command "#{python2} -m pip install --no-deps  #{windows_safe_path(project_dir)}\\datadog_checks_base"
+      command "#{python2} -m pip install --no-deps  #{windows_safe_path(project_dir)}\\datadog_checks_downloader --install-option=\"--install-scripts=#{windows_safe_path(install_dir)}/bin\""
+      command "#{python2} -m piptools compile --generate-hashes --output-file #{windows_safe_path(install_dir)}\\#{agent_requirements_file} #{static_reqs_out_file}"
```

Comment:
The resolving of dependencies is also influenced by the Python version so this (and other logic) should point to the proper Python and a version-specific resolved `agent_requirements_file`

Reply:
I don't have a clear idea of what the def should look like in order to support both Pythons yet, we will address these concerns in a dedicated PR to ease the review process